checks.d/mcache.py

```diff
@@ -170,22 +223,93 @@ def _get_metrics(self, server, port, tags):
                 tags=service_check_tags,
                 message="Server has been up for %s seconds" % uptime)
         except AssertionError:
-            self.service_check(
-                self.SERVICE_CHECK, AgentCheck.CRITICAL,
-                tags=service_check_tags,
-                message="Unable to fetch stats from server")
-            raise Exception(
-                "Unable to retrieve stats from memcache instance: {0}:{1}."
-                "Please check your configuration".format(server, port))
+            raise
 
-        if mc is not None:
-            mc.disconnect_all()
-            self.log.debug("Disconnected from memcached")
-        del mc
+
+    def _get_optional_metrics(self, client, tags, options=None):
+        for arg, metrics_args in self.OPTIONAL_STATS.iteritems():
+            if not options or options.get(arg, False):
```

Comment:
@degemer Not really, I supposed that if you called the method without setting the `options` dict, then you wanted all `options` (otherwise you wouldn't call the `_get_optional_metrics()` method). On the other hand if you do set the `options` dict, then only collect what is set to `True` there.

I enforce the logic you're thinking about in `check()` where we only call `_get_optional_metrics()` if `options` are set in the YAML.

Let me know if you disagree.